tests/custom_ekus.rs

```diff
@@ -0,0 +1,68 @@
+#[cfg(feature = "alloc")]
+fn check_cert(
+    ee: &[u8],
+    ca: &[u8],
+    eku: webpki::ExtendedKeyUsage,
+    result: Result<(), webpki::Error>,
+) {
+    let anchors = vec![webpki::TrustAnchor::try_from_cert_der(ca).unwrap()];
+    let anchors = webpki::TrustAnchors(&anchors);
+
+    let time = webpki::Time::from_seconds_since_unix_epoch(0x1fed_f00d);
+    let cert = webpki::EndEntityCert::try_from(ee).unwrap();
+
+    assert_eq!(
+        cert.verify_is_valid_cert_with_eku(
+            eku,
+            &[&webpki::RSA_PKCS1_2048_8192_SHA256],
+            &anchors,
+            &[],
+            time,
+            &[],
+        ),
+        result
+    );
+}
+
+#[cfg(feature = "alloc")]
+#[allow(clippy::identity_op)]
+static EKU_CLIENT_AUTH: webpki::KeyPurposeId =
+    webpki::KeyPurposeId::new(&[(40 * 1) + 3, 6, 1, 5, 5, 7, 3, 2]);
+
+#[cfg(feature = "alloc")]
+#[allow(clippy::identity_op)]
+static EKU_SERVER_AUTH: webpki::KeyPurposeId =
+    webpki::KeyPurposeId::new(&[(40 * 1) + 3, 6, 1, 5, 5, 7, 3, 1]);
+
+#[cfg(feature = "alloc")]
+#[test]
+pub fn verify_custom_eku() {
```

Comment:
This is a good test for the semantics of `Required`/`RequiredIfPresent`, but really we need an example certificate that demonstrates the real-world data you are using.

There is a real risk that, because this crate is made and maintained for just TLS uses, that we could unknowingly break it for your use case in the future. eg, by tightening up the validation done in `EndEntityCert::try_from`.

Reply:
+1, I think this is a good suggestion.

Reply:
Done. I took the example certificates from the ISO spec itself.